transaction/_transaction.py

```diff
@@ -760,3 +753,13 @@ def __init__(self, datamanager):
 
     def rollback(self):
         raise TypeError("Savepoints unsupported", self.datamanager)
+
+def text_or_warn(s):
+    if isinstance(s, text_type):
+        return s
+
+    warnings.warn("Expected text", DeprecationWarning, stacklevel=3)
+    if isinstance(s, bytes):
+        return s.decode('utf-8', 'replace')
+    else:
+        return text_type(s)
```

Comment:
No, nothing is ignored here. (None is ignored at the call sites.)

Anything but text generates a warning.  Bytes get special treatment by decoding them. Everything else gets "stred" by converting them to text. The later because on the transaction mailing list, somone asserted that Plone sometimes passes ints. <shrug>.